org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -181,7 +181,14 @@ public void disconnectClient() {
 	public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
 		logInfo(">> initialize");
 		InitHandler handler = new InitHandler(pm, preferenceManager, client);
-		return CompletableFuture.completedFuture(handler.initialize(params));
+		InitializeResult result = handler.initialize(params);
+		try {
+			jvmConfigurator.configureDefaultVM(preferenceManager.getPreferences());
```

Comment:
Fix NPE is ok, but don't get the point why need manually configure default vm.

Reply:
To update JAVA_HOME sent by the user as preference in the  initialize request instead of using the default runtime JVM.

Reply:
FYR: Please take a look at the "JVMConfigurator. configureDefaultVM" method

Link: https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JVMConfigurator.java#L45

Reply:
The JAVA_HOME is supposed to point to the same VM as the default runtime JVM. Could you view the latest comment for issue https://github.com/eclipse/eclipse.jdt.ls/issues/1068?

Reply:
Hi All,

Appoligize for the very late response.

What am actually trying to do is:

I want to set different JDK version for the project and i can't find any way to do that, so i thought to update the JAVA_HOME which is sent on initialize request.

Am calling the "java/buildWorkspace" request once the initialization is request is completed. But the JAVA_HOME is not getting changed while building the workspace and it is getting changed only while calling triggering "workspace/didChangeConfiguration" and so i thought to invoke the "JVMConfigurator.configureDefaultVM" method on initialize request too instead of invoking "workspace/didChangeConfiguration" separately.

Please let me know if there is any proper way to set JDK for a project.

Thanks,